bin/metric-zfs-arc.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+#
+#   ZFS ARC metrics
+#
+# DESCRIPTION:
+#   This collects ZFS ARC (Adaptive Replacement Cache) metrics
+#   metric-zfs-arc.rb looks at /proc/spl/kstat/zfs/arcstats
+#
+# OUTPUT:
+#  metric data
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: socket
+#
+# USAGE:
+#  ./metric-zfs-arc.rb
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2019 Airbrake Technologies, Inc <support@airbrake.io>
+#   Released under the same terms as the Sensu (the MIT license); see LICENSE
+#   for details.
+
+require 'sensu-plugin/metric/cli'
+require 'socket'
+
+#
+# ZFS Arc Metrics
+##
+class ZfsArcMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to .$parent.#child',
+         short: '-s SCHEME',
+         long: '--scheme SCHEME',
+         default: "#{Socket.gethostname}.zfs"
+
+  def run
+    if File.exist?('/proc/spl/kstat/zfs/arcstats')
+      command = 'cat /proc/spl/kstat/zfs/arcstats'
```

Comment:
rather than using cat its probably better to use native ruby implementation as it gives us better opportunities to rescue in the case of failure, for example just because the file exists does not mean its readable by the user requesting to read it. On that note it might be worth adding a check for this: https://ruby-doc.org/core-2.4.0/File.html#method-c-readable_real-3F

```suggestion
      command = File.read('/proc/spl/kstat/zfs/arcstats')
```

Reply:
:+1:

Reply:
I don't think we can use `File.read` because files in `/proc` aren't typically real files. i.e. `/proc` shows information relation to processes in a file hierarchy

Reply:
I'm pretty sure if `cat` works so will `File.read`, if it's a device your probably right but cat probably would not help you either. Can you open up an IRB or pry session to validate?
```
$ irb
irb(main):001:0> File.read('/proc/1/comm')
=> "systemd\n"
```

Reply:
Okay this cleans it up nicely

`File.read('/proc/spl/kstat/zfs/arcstats').split("\n").drop(2).each do |row|`